### PR TITLE
Feat/complete login

### PR DESCRIPTION
**Pull request by deevem**

### CODE REVIEW

File: backend/app/service/user.py [hunk 1]
```diff
@@ -6,10 +6,21 @@
 
 class UserService():
     def create_user(self, username, password):
-        print("start create")
+        pat = '^[a-zA-Z0-9_-]{4,16}$'
+        user = login.User.query.filter_by(username=username).first()
+        if user:
+            return False
```
Comment (adfwer233) [inline backend/app/service/user.py:12]: Other checks?

File: backend/app/api/user.py [hunk 2]
```diff
@@ -17,22 +17,35 @@
...
+            return {'code': 1, 'message': 'bad arguments'}
+        else:
+            try:
+                exist = service.find_user(username)
+                if exist:
+                    return {'code': 1, 'message': 'user exists'}
+                else:
+                    return {'code': 0, 'message': 'user not exist'}
+            except:
+                return {'code': 0, 'message': 'user not exist'}
```
Comment (adfwer233) [inline backend/app/api/user.py:31]: add comments interpreting  'code'

**Review by adfwer233 — APPROVED** [review]

File: backend/app/auth/views.py [hunk 3]
```diff
@@ -1,36 +1,29 @@
...
-        username = form.username.data
-        password = form.password.data
+    form = request.form.to_dict()
+    username = form['username']
+    password = form['password']
+    if username:
         user = User.query.filter_by(username=username).first()
         if user:
-            if user.validate_password(password):
+            if(user.validate_password(password)):
```
Comment (adfwer233) [inline backend/app/auth/views.py:18]: do not write if(...), use if ... instead

**Review by adfwer233 — APPROVED** [review]

File: frontend/src/components/LoginBox.vue [hunk 4]
```diff
@@ -71,9 +74,22 @@ const submitForm = (formEl: FormInstance | undefined) => {
   formEl.validate((valid) => {
     if (valid) {
       console.log("submit!");
-      router.replace({ name: "main", params: { username: ruleForm.username } });
+      axios
+        .post("/auth/login", qs.stringify(ruleForm))
+        .then(function (response) {
+          const code = response.data["code"];
+          if (code == "200") {
```
Comment (hvlgo) [inline frontend/src/components/LoginBox.vue:81]: use === instead

File: frontend/src/components/SignUpBox.vue [hunk 5]
```diff
@@ -58,7 +57,16 @@ const checkUsername = (rule: any, value: any, callback: any) => {
     if (!uPattern.test(value)) {
       callback(new Error("用户名规则, 4到16位 (字母，数字，下划线，减号)"));
     } else {
-      callback();
+      axios
+        .get("/api/user?username=" + ruleForm.username)
+        .then(function (response) {
+          const code = response.data["code"];
+          if (code == 1) {
```
Comment (hvlgo) [inline frontend/src/components/SignUpBox.vue:64]: use ===

File: frontend/src/components/SignUpBox.vue [hunk 6]
```diff
@@ -124,6 +132,26 @@ const submitForm = (formEl: FormInstance | undefined) => {
   formEl.validate((valid) => {
     if (valid) {
       console.log("submit!");
+      axios.post("/api/user", qs.stringify(ruleForm)).then(function (response) {
+        const code = response.data["code"];
+        if (code == "200") {
```
Comment (hvlgo) [inline frontend/src/components/SignUpBox.vue:137]: as above

File: frontend/src/components/SignUpBox.vue [hunk 7]
```diff
@@ -124,6 +132,26 @@ const submitForm = (formEl: FormInstance | undefined) => {
...
+          axios
+            .post("/auth/login", qs.stringify(ruleForm))
+            .then(function (response) {
+              const code = response.data["code"];
+              console.log(code);
+              if (code == "200") {
```
Comment (hvlgo) [inline frontend/src/components/SignUpBox.vue:143]: as above

**Review by hvlgo — COMMENTED** [review]
cannot test successfully, some bug in database. the sign up box also has bug that the "two input don't match" show incorrectly

**Review by hvlgo — APPROVED** [review]

**Review by hvlgo — APPROVED** [review]

**Pull request merged**